ext/odbc/php_odbc.c

```diff
@@ -2093,9 +2124,18 @@ bool odbc_sqlconnect(zval *zv, char *db, char *uid, char *pwd, int cur_opt, bool
 	memset(link->connection, 0, sizeof(odbc_connection));
 
 	link->persistent = persistent;
-	link->hash = hash;
-	SQLAllocEnv(link->connection->henv);
-	SQLAllocConnect(link->connection->henv, link->connection->hdbc);
+	link->hash = zend_string_copy(hash);
+	ret = SQLAllocEnv(&((*link->connection).henv));
+	if (ret != SQL_SUCCESS && ret != SQL_SUCCESS_WITH_INFO) {
```

Comment:
I added this check when debugging the above mentioned issue. I think we could keep not checking the error condition, and then an error would surface later on anyway, but I guess it's better to catch it as early as possible. I can remove this though if you think it's not needed.